app/util/security/LdapAuthenticationProvider.scala

```diff
@@ -32,76 +38,116 @@ class LdapAuthenticationProvider() extends AuthenticationProvider {
     Context.PROVIDER_URL -> url,
     Context.SECURITY_AUTHENTICATION -> "simple")
 
-  // returns uid=USERNAME,ou=people
-  protected def getPrincipal(username: String): String = {
-    "uid=%s,%s".format(username, usersub)
-  }
-
-  // returns uid=USERNAME,ou=people,dc=example,dc=org
-  protected def getSecurityPrincipal(username: String): String = {
-    "%s,%s".format(getPrincipal(username), searchbase)
-  }
-
-  // returns (&cn=*)(uniqueMember=uid=USERNAME,ou=people,dc=example,dc=org)
-  protected def groupQuery(username: String): String = {
+  protected def groupQuery(dn: String, username: String): String = {
     if (config.isRfc2307Bis)
-      "(&(cn=*)(%s=%s))".format(groupattrib, getSecurityPrincipal(username))
+      "(&(%s=*)(%s=%s))".format(groupnameattrib, groupattrib, dn)
```

Comment:
dn seems like a funny attribute to use for a group lookup.  I would think basedn would be better, but I'm not sure what value it is likely to have.  Do you expect dn == 'ou=people,dc=example,dc=org' or 'ou=group,dc=example,dc=org' ?